### PR TITLE
create ImageCollector play

### LINKED ISSUE #645: [Add a Play]: Image Collector
Thank You! 

## What's the Play about?
Users can collect images in a grid by submitting image URLs

## What ReactJS concept will be used to create this play?
useState, useEffect

## Are you willing to take it up for implementation?
Yes

### PR DESCRIPTION
**Pull request by AhnafAhamed**
# Description

Created the Image Collector Play.

Fixes #645 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@murtuzaalisurti plz review

**Comment by murtuzaalisurti** [comment]
> @murtuzaalisurti plz review

yeah

File: src/plays/image-collector/styles.css [hunk 1]
```diff
@@ -0,0 +1,51 @@
+.input-container {
+  width: max-content;
+  margin: auto;
+}
+
+.input-url {
+  padding: 8px 16px;
+  width: 400px;
```
Comment (murtuzaalisurti) [inline src/plays/image-collector/styles.css:8]: please make the input element responsive by using relative css units

**Review by murtuzaalisurti — CHANGES_REQUESTED** [review]
there are some changes that need to be made

**Review by koustov — CHANGES_REQUESTED** [review]
Early feedback. Need to maintain the image aspect ratio
image

**Comment by AhnafAhamed** [comment]
Thanks for the feedback @murtuzaalisurti @koustov. I have added the requested changes. Thank You

**Comment by murtuzaalisurti** [comment]
@AhnafAhamed the aspect ratio is still not correct for some images as mentioned by @koustov, maybe you can do `background-size: cover`.

here's some reference https://css-tricks.com/almanac/properties/b/background-size/

**Comment by AhnafAhamed** [comment]
> @AhnafAhamed the aspect ratio is still not correct for some images as mentioned by @koustov, maybe you can do `background-size: cover`.
> 
> here's some reference https://css-tricks.com/almanac/properties/b/background-size/

I have added `object-fit: cover;` to the image which is similar to `background-size` property. Do you want me to make this image a background-image and add the `background-size: cover;` property? Thanks

**Comment by murtuzaalisurti** [comment]
@atapas the vercel preview is not updating

**Review by murtuzaalisurti — APPROVED** [review]
looks fine

**Comment by atapas** [comment]
@AhnafAhamed Please take care of @koustov's comments.

**Comment by AhnafAhamed** [comment]
> @AhnafAhamed Please take care of @koustov's comments.

Hi @atapas 

I have already fixed it and @murtuzaalisurti also said it's fine. 

Below you can see I have added different images and there is no problem with the aspect ratio.


image

**Comment by atapas** [comment]
Thanks @AhnafAhamed 

@koustov please review

**Comment by koustov** [comment]
I still have some observation

1. Images are a bit scattered over the container. I mean these two images are way apart not properly center-aligned. 
2. Also the landscape images are cropped. Is it by design? 
3. The boxes should have bit of borders. Notice the PNG image borders are kind of vanished with the background 

image

image

File: src/plays/image-collector/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- Blog: null
+- Video: null
+
+## Implementation Details
+
+Update your implementation idea and details here
+
+## Consideration
+
+Update all considerations(if any)
```
Comment (koustov) [inline src/plays/image-collector/readme.md:23]: Remove the section if not applicable

File: src/plays/image-collector/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## Creator Information
+
+- User: AhnafAhamed
+- Gihub Link: https://github.com/AhnafAhamed
+- Blog: null
+- Video: null
+
+## Implementation Details
+
+Update your implementation idea and details here
```
Comment (koustov) [inline src/plays/image-collector/readme.md:19]: Give implementation details for your play

File: src/plays/image-collector/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+## Consideration
+
+Update all considerations(if any)
+
+## Resources
+
+Update external resources(if any)
```
Comment (koustov) [inline src/plays/image-collector/readme.md:27]: Remove the section if not applicable

**Review by koustov — CHANGES_REQUESTED** [review]

**Comment by AhnafAhamed** [comment]
Hi @koustov 

I have made improvements base on your suggestions, please review them.

Thank You for the feedback.

**Comment by AhnafAhamed** [comment]
Hi @koustov any updates on this?

File: src/plays/image-collector/ImageCollector.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import PlayHeader from "common/playlists/PlayHeader";
+import { useState } from "react";
+import "./styles.css";
+
+function ImageCollector(props) {
+  const [urlInput, setUrlInput] = useState("");
+  const [images, setImages] = useState([]);
+  // const images = useRef([]);
```
Comment (murtuzaalisurti) [inline src/plays/image-collector/ImageCollector.js:8]: can you remove this comment if it's not required?

**Review by murtuzaalisurti — CHANGES_REQUESTED** [review]

File: src/plays/image-collector/ImageCollector.js (hunk 5 above)
Comment (AhnafAhamed) [reply src/plays/image-collector/ImageCollector.js]: @murtuzaalisurti removed it

**Review by koustov — APPROVED** [review]
lgtm

@murtuzaalisurti is there anything pending from your end?

File: src/plays/image-collector/ImageCollector.js (hunk 5 above)
Comment (murtuzaalisurti) [reply src/plays/image-collector/ImageCollector.js]: great, looks good

**Comment by murtuzaalisurti** [comment]
> lgtm
> 
> @murtuzaalisurti is there anything pending from your end?

nope, all clear

**Review by murtuzaalisurti — APPROVED** [review]
looks good

**Pull request merged**